chained/water_impl.go

```diff
@@ -20,11 +25,36 @@ type waterImpl struct {
 	nopCloser
 }
 
+var httpClient *http.Client
+
 func newWaterImpl(addr string, pc *config.ProxyConfig, reportDialCore reportDialCoreFn) (*waterImpl, error) {
+	var wasm []byte
+
 	b64WASM := ptSetting(pc, "water_wasm")
-	wasm, err := base64.StdEncoding.DecodeString(b64WASM)
-	if err != nil {
-		return nil, fmt.Errorf("failed to decode water wasm: %w", err)
+	if b64WASM != "" {
+		var err error
+		wasm, err = base64.StdEncoding.DecodeString(b64WASM)
+		if err != nil {
+			return nil, fmt.Errorf("failed to decode water wasm: %w", err)
+		}
+	}
+
+	wasmAvailableAt := ptSetting(pc, "water_available_at")
+	if wasm == nil && wasmAvailableAt != "" {
+		urls := strings.Split(wasmAvailableAt, ",")
+		b := new(bytes.Buffer)
+		cli := httpClient
+		if cli == nil {
+			cli = proxied.ChainedThenDirectThenFrontedClient(1*time.Minute, "")
```

Comment:
These downloads may often take more than 1 minute.  Let's make this part of the proxy configuration or the global configuration.  I'll let you decide which, I think there are pros and cons to both.